contest/contest_info.py

```diff
@@ -110,6 +115,8 @@ def is_contestant(user,contest):
 
 def can_ask(user,contest):
     user_is_contestant = is_contestant(user,contest)
-    user_is_owner = (contest.owner == user)
-    user_is_coowner = contest.coowner.filter(pk = user.pk).exists()
-    return  user_is_contestant | user_is_owner | user_is_coowner 
+    user_is_owner_coowner = has_contest_ownership(user,contest)
+    return  user_is_contestant | user_is_owner_coowner
+
+def can_reply(user,contest):
```

Comment:
Ok, it's fine here. You remind me that there may have some bugs in ```has_xxx_ownership```.
@drowsy810301 maybe you should handle user authentication issue in ```has_xxx_ownership```